Add more ollama processor options (#2806)
redpanda-data/connect: docs/modules/components/pages/processors/ollama_chat.adoc

*Type*: `string`

=== `response_format`

> The response format of generated type, the model must also be prompted to output the appropriate response type.

Reviewer: ```suggestion
The format of the response that the Ollama model generates. For example, `text`. The `prompt` field value must match the `response format` to output the appropriate response type.
```
I'm not sure if this is what you mean? What would you need to include as prompt to generate a `text` response?

*Type*: `string`

*Default*: `"text"`

Options:
`text`
, `json`
.

=== `max_tokens`

> The maximum number of tokens to predict and output.

Reviewer: Does more tokens mean a better response, in terms of accuracy or speed?

*Type*: `int`

=== `temperature`

> The temperature of the model. Increasing the temperature will make the model answer more creatively.

Reviewer: Does this mean that the responses will be less accurate?
```suggestion
The temperature of the model. Increasing the temperature makes the model answer more creatively.
```

*Type*: `int`

=== `num_keep`

> Specify the number of tokens from the initial prompt to retain when the model resets its internal context. By default, this value is set to 4. Use -1 to retain all tokens from the initial prompt.

Reviewer: Under what circumstances does a reset occur? What does setting this value to `4` do?

*Type*: `int`

=== `seed`

> Sets the random number seed to use for generation. Setting this to a specific number will make the model generate the same text for the same prompt.

Reviewer: Can we give an example?

Author: Added an example

*Type*: `int`

=== `top_k`

> Reduces the probability of generating nonsense. A higher value (e.g. 100) will give more diverse answers, while a lower value (e.g. 10) will be more conservative.

Reviewer: ```suggestion
Reduces the probability of generating nonsense. A higher value, for example `100`, gives more diverse answers.  A lower value, for example `10` gives more conservative answers.
```

*Type*: `int`

=== `top_p`

> Works together with top-k. A higher value (e.g., 0.95) will lead to more diverse text, while a lower value (e.g., 0.5) will generate more focused and conservative text.

Reviewer: ```suggestion
Works together with the `top-k` value. A higher value, for example `0.95` generates more diverse text. A lower value, for example `0.5`, generates more focused and conservative text.
```
I'm not sure I understand what more diverse text means?

*Type*: `float`

=== `repeat_penalty`

> Sets how strongly to penalize repetitions. A higher value (e.g., 1.5) will penalize repetitions more strongly, while a lower value (e.g., 0.9) will be more lenient.

Reviewer: ```suggestion
Sets how strongly to penalize repetitions. A higher value, for example `1.5` will penalizes repetitions more strongly. A lower value, for example `0.9` is more lenient.
```

*Type*: `float`

=== `presence_penalty`

> Positive values penalize new tokens based on whether they appear in the text so far, increasing the model's likelihood to talk about new topics.

Reviewer: ```suggestion
Positive values penalize new tokens if they have appeared in the text so far. This increases the model's likelihood to talk about new topics.
```
Do we want to suggest a value as for the other fields/options above?

*Type*: `float`

=== `frequency_penalty`

> Positive values penalize new tokens based on their existing frequency in the text so far, decreasing the model's likelihood to repeat the same line verbatim.

Reviewer: ```suggestion
Positive values penalize new tokens based on the frequency of their appearance in the text so far. This decreases the model's likelihood to repeat the same line verbatim.
```
Do we want to suggest a value as for the other fields/options above?

*Type*: `float`

=== `stop`

Sets the stop sequences to use. When this pattern is encountered the LLM will stop generating text and return.

*Type*: `array`

=== `runner`

Options for the model runner that are used when the model is first loaded into memory.

*Type*: `object`

=== `runner.context_size`

> Sets the size of the context window used to generate the next token.

Reviewer: Can we suggest and value and the trade offs? I assume that the setting a larger context window has some downsides?

*Type*: `int`

=== `runner.batch_size`

The maximum number of requests to process in parallel.

*Type*: `int`

=== `runner.gpu_layers`

> This option allows offloading some layers to the GPU for computation. Generally results in increased performance. By default the runtime decides the number of layers dynamically.

Reviewer: ```suggestion
This option allows offloading some layers to the GPU for computation. This generally results in increased performance. By default, the runtime decides the number of layers dynamically.
```
Again, would it be useful to give an example?

Author: I highly suggest that users leave this as the default, as the runtime will offload as much as it can to GPU devices

*Type*: `int`

=== `runner.threads`

> Set the number of threads to use during generation. For optimal performance, it is recommended to set this value to the number of physical CPU cores your system has. By default the runtime decides the optimal number of threads.

Reviewer: ```suggestion
Set the number of threads to use during generation. For optimal performance, it is recommended to set this value to the number of physical CPU cores your system has. By default, the runtime decides the optimal number of threads.
```

*Type*: `int`

=== `runner.use_mmap`

> If supported, map the model into memory. This allows the system to load only the necessary parts of the model as needed.

Reviewer: Under what circumstances might this option not be supported?

*Type*: `bool`

=== `runner.use_mlock`

> Lock the model in memory, preventing it from being swapped out when memory-mapped. This can improve performance but trades away some of the advantages of memory-mapping by requiring more RAM to run and potentially slowing down load times as the model loads into RAM.

Reviewer: ```suggestion
Lock the model in memory, preventing it from being swapped out when memory-mapped. This option can improve performance but reduces some of the advantages of memory-mapping because it uses more RAM to run and can slow down load times as the model loads into RAM.
```
